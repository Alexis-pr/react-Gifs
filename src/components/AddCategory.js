import React, { useState } from 'react'
import PropTypes from 'prop-types';

const AddCategory = ( {setCategories}) => {
    const [inputValue, setInputValue] = useState('')

    const handleChange=(e)=>{
        setInputValue(e.target.value)
        console.log('HandleChange fue llamado desde addCategory')
    }
    const handleSubmit =(e)=>{
        e.preventDefault()
        console.log('esto es un llamado desde el handleSubmit de addCategory')
        if( inputValue.trim().length >= 3){
          setCategories(categor => [inputValue , ...categor ] )
          setInputValue('')
        } else{
            alert('Tiene q ser mayor a 3 letras')
        } 
            
    }

    return (
        <>
         {/*    <h2>Buscar...</h2>   */}
            <form onSubmit={handleSubmit}>
                <p>{inputValue}</p>
                <input 
                    type='text'
                    value={inputValue}
                    onChange={handleChange}
                    placeholder="Buscar...."
                />
            </form>
             
        </>
    )
}
AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}
export default AddCategory
