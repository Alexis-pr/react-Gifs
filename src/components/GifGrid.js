import React  from 'react'
import { UseFetchGifs } from '../hooks/UseFetchGifs'
import { GifGridItem } from './GifGridItem'

export const GifGrid = ({category}) => {
 /*    const [images, setImages] = useState([])
    useEffect(() => {
        getGifs(category)
         .then(setImages)
    }, [category])
      */
  const {data:images,loading} = UseFetchGifs( category);

    return (
        <>
         
            {loading && 'cargando...' }
            <h3 className="animate__animated animate__fadeIn animate__delay-2s">{category}</h3> <hr/> 
            <div className='card-grid'>
                {
                    images.map((img)=> (
                    <GifGridItem 
                        key={img.id}
                        {...img} /* Esto es igual a { img.url , img.title} tambien se puede usar el destructurado pero esto tomaria mas lineas de codigo
                        mientras que de esta forma es mas facil y legible el codigo  */
                        />))
                }
            </div>
        </>

    )
}
