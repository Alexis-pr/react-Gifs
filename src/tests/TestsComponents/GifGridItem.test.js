import '@testing-library/react'
import  {GifGridItem} from './../../components/GifGridItem'
import React from 'react'
import { shallow } from "enzyme" 



describe('Pruebas del Componente GifGridItem.js', () => {
    const title = 'Hola de titulo'
    const url = 'https://localhost/algo.jpg'
    const wrapper = shallow( <GifGridItem title={title} url={url} /> )

    test('Debe de mostrar el componente correctamente ', () => {
        expect(wrapper).toMatchSnapshot() 
    })

    test('Debe de tener un texto el title', () => {
        const p = wrapper.find('h4')
        expect( p.text().trim() ).toBe(title);
    })
    
    test('La imagen debe de tener las dos propiedades correctamente', () => {
        const img = wrapper.find('img')
        /* console.log(img.props().src) */
        expect(img.prop('src')).toBe(url)
        expect( img.prop('alt')).toBe(title)
    })

    test('El div debe de tener la clase correcta', () => {
        const cont = wrapper.find('div')
        /*  console.log( cont.prop('className') )  */
        const className = cont.prop('className')
        expect(className.includes('animate__fadeInDown')).toBe(true)
        
    })
    
    
    
})
