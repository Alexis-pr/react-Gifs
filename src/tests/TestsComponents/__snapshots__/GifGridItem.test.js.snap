// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas del Componente GifGridItem.js Debe de mostrar el componente correctamente  1`] = `
<div
  className="card animate__animated animate__fadeInDown "
>
  <img
    alt="Hola de titulo"
    src="https://localhost/algo.jpg"
  />
  <h4>
    Hola de titulo
  </h4>
</div>
`;
